tx-pool/src/service.rs

```diff
@@ -0,0 +1,620 @@
+        let (responder, response) = crossbeam_channel::bounded(1);
+        let request = Request::call(id, responder);
+        sender.try_send(Message::FetchTxRPC(request))?;
+        response.recv().map_err(Into::into)
+    }
+
+    pub fn fetch_txs(
+        &self,
+        short_ids: Vec<ProposalShortId>,
+    ) -> Result<HashMap<ProposalShortId, TransactionView>, FailureError> {
+        let mut sender = self.sender.clone();
+        let (responder, response) = crossbeam_channel::bounded(1);
+        let request = Request::call(short_ids, responder);
+        sender.try_send(Message::FetchTxs(request))?;
+        response.recv().map_err(Into::into)
+    }
+
+    pub fn fetch_txs_with_cycles(
+        &self,
+        short_ids: Vec<ProposalShortId>,
+    ) -> Result<FetchTxsWithCyclesResult, FailureError> {
+        let mut sender = self.sender.clone();
+        let (responder, response) = crossbeam_channel::bounded(1);
+        let request = Request::call(short_ids, responder);
+        sender.try_send(Message::FetchTxsWithCycles(request))?;
+        response.recv().map_err(Into::into)
+    }
+}
+
+impl Drop for TxPoolController {
+    fn drop(&mut self) {
+        self.stop.try_send();
+    }
+}
+
+pub struct TxPoolServiceBuiler {
+    service: Option<TxPoolService>,
+}
+
+impl TxPoolServiceBuiler {
+    pub fn new(
+        tx_pool_config: TxPoolConfig,
+        snapshot: Arc<Snapshot>,
+        block_assembler_config: Option<BlockAssemblerConfig>,
+        txs_verify_cache: Lock<LruCache<Byte32, Cycle>>,
+        snapshot_mgr: Arc<SnapshotMgr>,
+    ) -> TxPoolServiceBuiler {
+        let last_txs_updated_at = Arc::new(AtomicU64::new(0));
+        let tx_pool = TxPool::new(tx_pool_config, snapshot, Arc::clone(&last_txs_updated_at));
+        let block_assembler = block_assembler_config.map(BlockAssembler::new);
+
+        TxPoolServiceBuiler {
+            service: Some(TxPoolService::new(
+                tx_pool,
+                block_assembler,
+                txs_verify_cache,
+                last_txs_updated_at,
+                snapshot_mgr,
+            )),
+        }
+    }
+
+    pub fn start(mut self) -> TxPoolController {
+        let (signal_sender, signal_receiver) = oneshot::channel();
+        let (sender, receiver) = mpsc::channel(DEFAULT_CHANNEL_SIZE);
+
+        let thread_builder = thread::Builder::new().name("TX-POOL".to_string());
+        let service = self.service.take().expect("tx pool service start once");
+
+        let thread = thread_builder
+            .spawn(move || {
+                let server = receiver
+                    .for_each(move |message| {
+                        let service_clone = service.clone();
+                        tokio::spawn(service_clone.process(message))
+                    })
+                    .select2(signal_receiver)
+                    .map(|_| ())
+                    .map_err(|_| ());
+                tokio::run(server);
```

Comment:
Do we need to share a runtime with the network, or do we have one each like this?

Reply:
We have discussed it before, network uses own standalone runtime, other services share global one.
But now, we only have one service use Tokio, Default global runtime is fine for now. Thoughts?

Reply:
ok